lib/node_types/esp/src/main.cpp

```diff
@@ -936,7 +1241,53 @@ const unsigned long performance_interval = 5000; // Interval of 5 seconds in mil
 unsigned long performance_iteration_count = 0;
 double performance_total_execution_time = 0; // Total execution time in microseconds
 
-// main loop managing the cooperative multitasking
+/**
+ * @brief Arduino main loop - runs continuously
+ * 
+ * MAIN LOOP OPERATION
+ * ===================
+ * This is where IoTempower's cooperative multitasking happens. The loop manages:
+ * 
+ * 1. PRECISION vs REGULAR INTERVALS
+ *    - Precision interval: Very tight timing for sensitive devices (e.g., LED strips)
```

Comment:
don't use leds as example, but power consumption measurement (led timing is actually handled well in NeoPixelBus library)